WooCommerce/Classes/ViewRelated/FilterTabBar/TabbedViewController.swift

```diff
@@ -102,8 +102,21 @@ class TabbedViewController: UIViewController {
             return
         }
         tabBarStackView.removeArrangedSubview(customTabBarView)
+        customTabBarView.removeFromSuperview()
         self.customTabBarView = nil
     }
+
+    func appendToTabBar(_ tab: TabbedItem) {
+        // Setup child view controller
+        items.append(tab)
+        tabBar.items = items
+        configureChildViewControllers()
```

Comment:
1. I fear that calling this method will add all the viewControllers to the stackView again. (without removing existing views)
2. I also noticed that the custom range view controller is not refreshed/reloaded when added to the tab bar. 

Did you explore the path of rebuilding the `TabbedViewController` again instead of appending a `TabbedItem`. I am asking because I believe that approach will be simpler and less prone to unexpected bugs.

Reply:
Thanks for this.

I get the suggestion for point 1. I'll figure out a way to only add the new view controller instead.

Can you explain more about rebuilding the `TabbedViewController`?  I'm asking because `StoreStatsAndTopPerformersViewController`, which has the `createCustomRangeTab()` function, is a subclass of `TabbedViewController`. I'm not sure if you meant we have to rebuild `StoreStatsAndTopPerformersViewController` in order to add the extra tab.


As for this:
> I also noticed that the custom range view controller is not refreshed/reloaded when added to the tab bar.

I intentionally did not take a look at the content displaying behavior yet in this PR, hence why the tab item basically shows and does nothing. My thought is that this should be part of https://github.com/woocommerce/woocommerce-ios/issues/11985

Reply:
> I intentionally did not take a look at the content displaying behavior yet in this PR, hence why the tab item basically shows and does nothing. I haven't checked how the refresh/reload is being done. My thought is that this should be part of https://github.com/woocommerce/woocommerce-ios/issues/11985

I think it will be better to have an idea about how we are going to handle the lifecycle (load, reload) of this custom tab in this PR. We can handle the actual content building in https://github.com/woocommerce/woocommerce-ios/issues/11985. What do you think? 

I didn't check about how the `StoreStatsAndTopPerformersViewController` works. I suggested recreating `TabbedViewController ` as I thought it would be simpler. It was just a suggestion and if you think we can make the content display and reload behaviour of the custom range work without recreating `StoreStatsAndTopPerformersViewController` please feel free to follow the existing approach. 👍

Reply:
> What do you think?

Yup, I'll handle the lifecycle here.

Reply:
I added two relevant commits:

- https://github.com/woocommerce/woocommerce-ios/pull/11977/commits/61f084696817283c675c78a6b15a9a5479710faf This loads the data right after tab creation
- https://github.com/woocommerce/woocommerce-ios/pull/11977/commits/7dd01ae6431772a8094995fed89f604a66d066ef includes the custom tab into the `observeSelectedTimeRangeIndex()`, which does a re-sync on tab change if needed.

Additional note: I observed that in the first commit above, the data loading works. However, on the second commit for the re-sync,  it always goes to `.failure` inside `self.dashboardViewModel.syncSiteVisitStats()`. 

I have not checked deeply yet why this is the case, and I think it should be handled properly in a separate PR to check the backend handling. The commits above are just to ensure that the custom tab is included inside the sync and re-sync process just like the other tabs.

Reply:
> I have not checked deeply yet why this is the case, and I think it should be handled properly in a separate PR to check the backend handling.

Yes, I agree. Could you please log a sub-task so that we don't miss this?

Reply:
Added the task in https://github.com/woocommerce/woocommerce-ios/issues/12014